go/common/sgx/pcs/tcb.go

```diff
@@ -158,35 +170,38 @@ type TCBInfo struct {
 	FMSPC                   string     `json:"fmspc"`
 	PCEID                   string     `json:"pceId"`
 	TCBType                 int        `json:"tcbType"`
-	TCBEvaluationDataNumber int        `json:"tcbEvaluationDataNumber"`
+	TCBEvaluationDataNumber uint32     `json:"tcbEvaluationDataNumber"`
```

Comment:
need citation, or I must suspect an adversary is trying to make `-1` sneak by the `< Min` check :laughing:

Reply:
Citation for what? Unfortunately it is only defined as an `integer` in JSON. But in practice (current value is 12 and it is monotonically increasing) it can only be positive and this makes it easier to compare with the TCB evaluation data number set in the policy (which we control and enforce to be uint32 as we want explicit lengths in serializable structures).

Reply:
https://json-schema.org/draft/2020-12/json-schema-validation.html#name-type this looks like json schema, and json schema doesn't constrain it to be nonnegative.

does go have a signed int32 type that we could use?

Reply:
Yeah based on the JSON schema alone it can be negative, but given that the spec defines it as a monotonically increasing number and its latest value is 12, it will never be negative in practice and I believe it is fine if we reject negative values during parsing (which the JSON parser will do in case the destination type is an `uint32`).

Other `int` fields should maybe also be changed to more explicit types.

Reply:
ok